central/centralhealth/service/service_impl.go

```diff
@@ -79,15 +79,24 @@ func (s *serviceImpl) GetUpgradeStatus(ctx context.Context, empty *v1.Empty) (*v
 			requiredBytes := int64(math.Ceil(float64(currentDBBytes) * (1.0 + capacityMarginFraction)))
 
 			var toBeFreedBytes int64
-			if pgadmin.CheckIfDBExists(adminConfig, migrations.PreviousDatabase) {
+			exists, err := pgadmin.CheckIfDBExists(adminConfig, migrations.PreviousDatabase)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Failed to determine if %s database exists", migrations.PreviousDatabase)
+			}
+			if exists {
 				toBeFreedBytes, err = pgadmin.GetDatabaseSize(adminConfig, migrations.GetPreviousClone())
 				if err != nil {
 					return nil, errors.Wrapf(err, "Fail to get database size %s", migrations.PreviousDatabase)
 				}
 
 				// Get a short-lived connection for the purposes of checking the version of the previous clone.
-				pool := pgadmin.GetClonePool(adminConfig, migrations.GetPreviousClone())
-				defer pool.Close()
+				pool, err := pgadmin.GetClonePool(adminConfig, migrations.GetPreviousClone())
+				if err != nil {
+					log.Infof("Unable to get previous database, leaving ForceRollbackTo empty.  %v", err)
```

Comment:
I'm not sure I understand the implication of this

Reply:
After looking at it, this should probably return an error.  At this point, a previous exists, but if for some reason when we try to connect to it to get the version we cannot.  If we simply log the error and return an empty version for `ForceRollbackTo` that would be very misleading to the caller.  I will change it to return an error